system_metrics_collector/src/system_metrics_collector/utilities.cpp

```diff
@@ -48,6 +59,102 @@ std::string readFileToString(const std::string & file_name)
   return to_return;
 }
 
+ProcCpuData processStatCpuLine(const std::string & stat_cpu_line)
+{
+  ProcCpuData parsed_data;
+
+  if (!stat_cpu_line.empty()) {
+    if (!stat_cpu_line.compare(0, strlen(CPU_LABEL), CPU_LABEL)) {
+      std::istringstream ss(stat_cpu_line);
+
+      if (!ss.good()) {
+        return ProcCpuData();
+      }
+      ss >> parsed_data.cpu_label;
+
+      for (int i = 0;
+        i < static_cast<int>(ProcCpuStates::kNumProcCpuStates); ++i)
+      {
+        if (!ss.good()) {
+          return ProcCpuData();
+        }
+        ss >> parsed_data.times[i];
+      }
+      return parsed_data;
+    }
+  }
+  return parsed_data;
+}
+
+ProcPidCpuData processPidStatCpuLine(const std::string & stat_cpu_line)
+{
+  ProcPidCpuData parsed_data;
+
+  if (!stat_cpu_line.empty()) {
+    std::istringstream ss(stat_cpu_line);
+    ss.exceptions(std::ios::failbit | std::ios::badbit);
+
+    ss.ignore(std::numeric_limits<std::streamsize>::max(), ' ');  // pid
+    ss.ignore(std::numeric_limits<std::streamsize>::max(), ' ');  // comm
+    ss.ignore(std::numeric_limits<std::streamsize>::max(), ' ');  // state
+    ss.ignore(std::numeric_limits<std::streamsize>::max(), ' ');  // ppid
+    ss.ignore(std::numeric_limits<std::streamsize>::max(), ' ');  // pgrp
+    ss.ignore(std::numeric_limits<std::streamsize>::max(), ' ');  // session
+    ss.ignore(std::numeric_limits<std::streamsize>::max(), ' ');  // tty_nr
+    ss.ignore(std::numeric_limits<std::streamsize>::max(), ' ');  // tpgid
+    ss.ignore(std::numeric_limits<std::streamsize>::max(), ' ');  // flags
+    ss.ignore(std::numeric_limits<std::streamsize>::max(), ' ');  // minflt
+    ss.ignore(std::numeric_limits<std::streamsize>::max(), ' ');  // cminflt
+    ss.ignore(std::numeric_limits<std::streamsize>::max(), ' ');  // majflt
+    ss.ignore(std::numeric_limits<std::streamsize>::max(), ' ');  // cmajflt
+    ss >> parsed_data.utime;
+    ss >> parsed_data.stime;
+
+    if (!ss.good()) {
```

Comment:
I don't think you need this anymore given you set the exception throwing bits.

Reply:
Removed.